pyrefinebio/dataset.py

```diff
@@ -0,0 +1,151 @@
+        self.is_available = is_available
+        self.has_email = has_email
+        self.email_address = email_address
+        self.email_ccdl_ok = email_ccdl_ok
+        self.expires_on = expires_on
+        self.s3_bucket = s3_bucket
+        self.s3_key = s3_key
+        self.success = success
+        self.failure_reason = failure_reason
+        self.created_at = created_at
+        self.last_modified = last_modified
+        self.start = start
+        self.size_in_bytes = size_in_bytes
+        self.sha1 = sha1
+        self.quantile_normalize = quantile_normalize
+        self.quant_sf_only = quant_sf_only
+        self.svd_algorithm = svd_algorithm
+        self.download_url = download_url
+
+    @classmethod
+    def create(
+        cls,
+        data,
+        aggregate_by=None,
+        scale_by=None,
+        email_address=None,
+        email_ccdl_ok=None,
+        start=None,
+        quantile_normalize=None,
+        quant_sf_only=None,
+        svd_algorithm=None
+    ):
+        body = {}
+        body["data"] = data
+        if aggregate_by:
+            body["aggregate_by"] = aggregate_by
+        if scale_by:
+            body["scale_by"] = scale_by
+        if email_address:
+            body["email_address"] = email_address
+        if email_ccdl_ok:
+            body["email_ccdl_ok"] = email_ccdl_ok
+        if start:
+            body["start"] = start
+        if quantile_normalize:
+            body["quantile_normalize"] = quantile_normalize
+        if quant_sf_only:
+            body["quant_sf_only"] = quant_sf_only
+        if svd_algorithm:
+            body["svd_algorithm"] = svd_algorithm
+            
+        response = post_by_endpoint("dataset", payload=body)
+        return Dataset(**response)
+
+    @classmethod
+    def get(cls, id):
+        response = get_by_endpoint("dataset/" + id)
+        return Dataset(**response)
+
+    def update(
+        self,
+        data,
+        aggregate_by=None,
+        scale_by=None,
+        email_address=None,
+        email_ccdl_ok=None,
+        start=None,
+        quantile_normalize=None,
+        quant_sf_only=None,
+        svd_algorithm=None
+    ):
+        body = {}
+        body["data"] = data
+        if aggregate_by:
+            body["aggregate_by"] = aggregate_by
+        if scale_by:
+            body["scale_by"] = scale_by
+        if email_address:
+            body["email_address"] = email_address
+        if email_ccdl_ok:
+            body["email_ccdl_ok"] = email_ccdl_ok
+        if start:
+            body["start"] = start
+        if quantile_normalize:
+            body["quantile_normalize"] = quantile_normalize
+        if quant_sf_only:
+            body["quant_sf_only"] = quant_sf_only
+        if svd_algorithm:
+            body["svd_algorithm"] = svd_algorithm
+
+        response = put_by_endpoint("dataset/" + self.id, payload=body)
+        return Dataset(**response)
+
+    def process(self, email_address):
```

Comment:
I think email_address should be optional as long as `self.email_address is not None`.